mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoMethodGenerator.java

```diff
@@ -146,6 +146,13 @@ protected void populateBuilderWithFunction(
     }
   }
 
+  protected void populateBuilderWithProfile(CodeBlock.Builder builder) {
+    builder.beginControlFlow("if(context.getExecutionProfileName() !=null)");
+    builder.addStatement(
+        "boundStatementBuilder = boundStatementBuilder.setExecutionProfileName(context.getExecutionProfileName())");
```

Comment:
are we handling `SimpleStatement`s as well? (I see logic only for boundStatements)

Reply:
The simple statements you are seeing are during the Dao initialization, these are prepared, however they are never executed directly. The prepared statements are cached in the DAO, and retrieved later to make bound statements then executed. I believe this is working correctly. If you can find a specific counter example please let me know... I couldn't find any instances of them being executed directly.